common/persistence/historyStore.go

```diff
@@ -481,12 +479,13 @@ func (m *historyV2ManagerImpl) readHistoryBranch(
 
 	nextPageToken, err := m.serializeToken(token)
 	if err != nil {
-		return nil, nil, nil, 0, 0, err
+		return nil, nil, nil, 0, 0, 0, err
 	}
 
-	return historyEvents, historyEventBatches, nextPageToken, dataSize, lastFirstEventID, nil
+	return historyEvents, historyEventBatches, nextPageToken, dataSize, lastFirstEventID, token.LastEventID, batchProcessingError
```

Comment:
What does it mean to return "nextPageToken" if batchProcessingError is not-nil? Can this lead to some weird scenario from the caller perspective?

Reply:
I think you are right, we should return nil token in case of an error, otherwise further iteration doesn't make any sense. Let me change that.